File: src/julia/partition/partition.jl
module Partition

export partition_sum
function partition_sum(arr::Vector{Int64}, size::Int, index::Int64) 
    sum = 0;

    for i in 1:size
        if index & (1 << (i - 1)) != 0
            sum += arr[i]
        end
    end

    return sum
end # partition_sum

end # module

File: src/go/partition/lib/partition.go
package partition

func Partition_sum(arr []int, index int64) int {
	var sum int = 0

	for i := 0; i < len(arr); i++ {
		if index&(1<<i) != 0 {
			sum += arr[i]
		}
	}

	return sum
}
